core-bundle/src/Resources/contao/dca/tl_undo.php

```diff
@@ -217,16 +225,23 @@ public function showDeletedRecords($data, $row)
 		return $data;
 	}
 
-	/**
-	 * Add the surrounding ellipsis layer
-	 *
-	 * @param array  $row
-	 * @param string $label
-	 *
-	 * @return string
-	 */
-	public function ellipsis($row, $label)
+	public function getFromTableOptions(DataContainer $dc)
 	{
-		return '<div class="ellipsis">' . $label . '</div>';
+		$tables = $this->Database->execute('SELECT DISTINCT ' . Database::quoteIdentifier('fromTable') . ' FROM tl_undo');
```

Comment:
Any reason why you're using the old `Database` class here instead of `System::getContainer()->get('connection')`?

Reply:
Nope, except that I just went with the legacy code flow ;-)